### PR TITLE
fix(angular): allow default browser action on CTRL+Click

### LINKED ISSUE #24413: bug: angular, routerLink prevents opening link in new tab
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [x] v4.x
- [X] v5.x
- [X] v6.x

### Current Behavior

When importing the IonicModule and using ``[routerLink]``,
the following line prevents Ctrl+Click behavior on Desktop browsers.

https://github.com/ionic-team/ionic-framework/blob/732f8e10ce604f1a3e98518ae9c3a4afd7803e9a/angular/src/directives/navigation/router-link-delegate.ts#L56

### Expected Behavior

Ctrl+Click should trigger the default browser behavior (e.g. open the link in a new tab/page/etc).

### Steps to Reproduce

1. Create an anchor element with a ``[routerLink]="[<your link>]"``
2. Ctrl+Click on it

### Code Reproduction URL

[Link to repository](https://github.com/Juarrow/ionic-angular-ctr-click-bug)

On the home page CTRL+Click the link

### Ionic Info

```
Ionic:

   Ionic CLI                     : 6.18.1 (C:\Users\Julian\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.0.1
   @angular-devkit/build-angular : 13.1.1
   @angular-devkit/schematics    : 13.0.3
   @angular/cli                  : 13.0.3
   @ionic/angular-toolkit        : 5.0.3

Utility:

   cordova-res : not installed globally
   native-run  : 1.5.0

System:

   NodeJS : v16.13.0 (C:\Program Files\nodejs\node.exe)
   npm    : 8.1.0
   OS     : Windows 10
```

### Additional Information

I'm currently monkey-patching it 🙈 .

Maybe this can be fixed by something like (pseudo-code)
````
if(platform is desktop && (key == CTRL || key == Meta) && button == 0)) {
 return true;
}
````

### PR DESCRIPTION
**Pull request by Juarrow**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #24413

## What is the new behavior?

- Allow CTRL/Meta + Click to be handled by the browser (e.g. for opening a link in a new tab)

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Juarrow** [comment]
1. Do i have to add to add changes in package-lock.json 's too? (that's why i did not tick that checkmark)
2. I kept the "old" behavior for ``<ion-button>``. I think this makes more sense for something that is supposed to be a button. What is your opinion?

**Comment by sean-perkins** [comment]
Hello @Juarrow thanks for this PR!

You do not need to commit `package-lock.json` changes. These are likely a result of running `npm i` instead of `npm ci` when cloning/setting-up the project. `ci` will install versions from the lock, where as `install` will attempt to install any new package versions based on the format in the package.json.

I think maintaining the existing `ion-button` behavior is appropriate. We can always make that change in a new minor release.

File: angular/src/directives/navigation/router-link-delegate.ts [hunk 1]
```diff
@@ -52,6 +54,16 @@ export class RouterLinkDelegateDirective implements OnInit, OnChanges, OnDestroy
    */
   @HostListener('click', ['$event'])
   onClick(ev: UIEvent): void {
+    if (!this.elementRef.nativeElement.closest('ion-button')) {
+      // In case of CTRL+Click or Meta+Click,
+      // ignore those to allow default browser actions.
+      if (this.platform.is('desktop') && ev instanceof MouseEvent) {
```
Comment (liamdebeasi) [inline angular/src/directives/navigation/router-link-delegate.ts:60]: Why do we need to check if the device is on desktop/a MouseEvent?

Comment (sean-perkins) [reply angular/src/directives/navigation/router-link-delegate.ts:60]: Agreed this appears to be redundant. The native `click` event is a [`MouseEvent`](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event). I'd advocate we remove the platform check to additionally handle tablet with a connected keyboard & mouse.

Comment (Juarrow) [reply angular/src/directives/navigation/router-link-delegate.ts:60]: Oftentimes, I tend to be too defensive with all the checks.

Will remove the if-statement.

Comment (sean-perkins) [reply angular/src/directives/navigation/router-link-delegate.ts:60]: @Juarrow let me know when you have a moment to update the PR with this change.

Comment (Juarrow) [reply angular/src/directives/navigation/router-link-delegate.ts]: @sean-perkins just did. Also updated the commit message scope to reflect your change to the PR title.

Comment (Juarrow) [reply angular/src/directives/navigation/router-link-delegate.ts]: @sean-perkins @liamdebeasi anything happening?

Comment (sean-perkins) [reply angular/src/directives/navigation/router-link-delegate.ts]: @Juarrow this task is in my sprint to both finish out and address this PR. The main issue effects each framework output. After looking deeper at `ion-button`, I would like to address to inconsistencies with that component (no action required on your end). 

Here's my WIP branch: https://github.com/ionic-team/ionic-framework/tree/FW-432 that pulls in your Angular change and applies a similar resolution to React and Vue. 

Once I am able to diagnose the `ion-button` behavior fully and add respective tests, I will likely open a new PR and give you co-author credit on the PR.

**Comment by sean-perkins** [comment]
Closing this PR in favor of: https://github.com/ionic-team/ionic-framework/pull/25014. I did give you co-authored-by credit for the commit. Appreciate the help here! Community PR contributions is still an area we are actively trying to improve on. We are trying to tackle the older PRs first, as they are often times the hardest PRs to evaluate. Our time to reply can definitely be improved 👍

**Pull request closed without merging**